pkg/stub/handler.go

```diff
@@ -81,16 +84,12 @@ type Handler struct {
 	sdkwrapper SDKWrapper
 	cvowrapper *operatorstatus.CVOOperatorStatusHandler
 
-	samplesResource *v1alpha1.SamplesResource
-	registrySecret  *corev1.Secret
-
-	// we maintain a separate SamplesResource cache
-	// entry when this is a default rhel install and we
-	// are waiting for the credential since we are
-	// in an erorr state; our current approach is
-	// to only update the standard samplesResource
-	// cache entry for valid / non-error states
-	waitingForCredential *v1alpha1.SamplesResource
+	configActive   bool
+	registrySecret *corev1.Secret
+	// This signifies that this is rhel install and we
+	// are waiting for the credential since they are
+	// required to access registry.redhat.io
+	waitingForCredential bool
```

Comment:
We should be able to determine this simply by looking at the SamplesResource event we recieved in the handler.

1) as long as the SamplesResource points to a registry that needs credentials, and we don't have credentials, and SamplesExist is false, requeue the event.

2) if the SamplesResource points to a registry that needs credentials and status.CredentialsExist==true, proceed with creating the samples

Reply:
I'm getting rid of `h.waitingForCredential` and `h.registrySecret` ...  bit more swizzling but I think worth it